src/main/java/io/syndesis/qe/marketplace/openshift/OpenShiftService.java

```diff
@@ -203,4 +212,75 @@ public void patchGlobalSecrets(String pullSecretContent) {
     public OpenShift getClient() {
         return openShiftClient;
     }
+
+    /**
+     * Retrieve the list of packages grouped by catalog sources.
+     * @param forceReload boolean, if the API should be always invoked and so the in-memory cache is ignored.
+     * @return {@link Map} collection that contains catalog source name (i.e. String 'redhat-operators') as key
+     * and {@link List} of {@link PackageManifest} as value.
+     */
+    public Map<String, List<PackageManifest>> getCatalog(boolean forceReload) {
+        if (catalog == null || forceReload) {
+            catalog = loadCatalog();
+        }
+        return catalog;
+    }
+
+    /**
+     * Invoke API to retrieve all CRD PackageManifest on 'openshift-marketplace' namespace
+     * @return {@link #catalog}
+     */
+    private Map<String, List<PackageManifest>> loadCatalog() {
+        AtomicReference<Map<String, List<PackageManifest>>> found = new AtomicReference<>();
+        CustomResourceDefinitionContext crds = new CustomResourceDefinitionContext.Builder()
+                .withVersion("v1")
+                .withKind("PackageManifest")
+                .withGroup("packages.operators.coreos.com")
+                .withScope("Namespaced")
+                .withPlural("packagemanifests")
+                .build();
+
+        final ObjectMapper mapper = new ObjectMapper();
+
+        openShiftClient.customResource(crds).list("openshift-marketplace").entrySet()
+                .stream().filter(entry -> "items".equals(entry.getKey())).findFirst().ifPresent( entry -> {
+                        found.set(((List<Map>) entry.getValue()).stream().collect(
+                                Collectors.groupingBy(pack -> (String) ((Map) pack.get("status")).get("catalogSource"),
+                                        HashMap::new,
+                                        Collectors.mapping(item -> mapper.convertValue(item.get("status"), PackageManifest.class), Collectors.toList())
+                                )));
+
+                    }
+                );
```

Comment:
This could perhaps be more readable if JSONObject was used instead
ie.
```
new JSONObject(openShiftClient.customResource(crds).list("openshift-marketplace")).getJSONArray("items").forEach(...)
```

Reply:
yes @mmuzikar , we can use JSONObject as well, thanks

Reply:
changes made, now the code is more readable even if the execution time is more or less 5 time slowest than using Java Stream APIs and Map object (using JSONObject I get ~200 mills vs ~40 mills using Map)